pkg/tls/tls.go

```diff
@@ -187,8 +190,7 @@ func (p *Provider) RootCA() []byte {
 // fails, returns error.
 func (p *Provider) fetchCertificate(ctx context.Context) error {
 	opts := pkiutil.CertOptions{
-		// TODO: allow configurable namespace and service
-		Host:       "cert-manager-istio-csr.cert-manager.svc",
+		Host:       strings.Join(p.dnsNames, ","),
```

Comment:
We should add a validation check for this config field in [`Complete()`](https://github.com/cert-manager/istio-csr/blob/65407367362d369e13683bd554b07077ce48a459/cmd/app/options/options.go#L105). We should error if the list is empty to exit early.